### ISSUE TITLE
Analysis is based on files on the disk, it should be based on in-memory content

### ISSUE THREAD

**Issue by loic-joly-sonarsource** [SonarSource/sonarlint-visualstudio#2084]
### Description

SonarLint analyses the files based on the content of disk. So if some files are not saved, it may analyze code in an inconsistent state, leading to spurious parsing error messages

### Repro steps

Please provide the steps required to reproduce the problem

1. Open the following project:
[SLVSInMemory.zip](https://github.com/SonarSource/sonarlint-visualstudio/files/5809582/SLVSInMemory.zip)

2. Modify the file header.h in vs, so that it contains:
```c++
#pragma once

int f(int i, int j) {
  return i+j;
}
```
Don't save it
3. Switch back to `SLVSInMemory.cpp`
Note that intellisense displays an error, which means it takes the modification into account. Change the line to contain:
```c++
    std::cout << "Hello World! " << f(40, 2);
```
The error from intellisense disappears.
4. Save this file.
SonatLint will display the message:
![image](https://user-images.githubusercontent.com/37904001/104474361-39650100-55be-11eb-823f-d279ae0d8f90.png)

If you go back to `header.h` and save it, then come back to `SLVSInMemory.cpp`, it will still not work correctly, you then have to do a meaningless change in this file (add a blank line), save it again, and the error disappears.

### Expected behavior

The in-memory content of the file is used for analysis

### Actual behavior

The on-disk status is used

**Comment by pavel-mikula-sonarsource** [comment]
Moved to [SLVS-2260](https://sonarsource.atlassian.net/browse/SLVS-2260)

[SLVS-2260]: https://sonarsource.atlassian.net/browse/SLVS-2260?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
